readthedocs/projects/tasks.py

```diff
@@ -553,6 +569,10 @@ def run_build(self, docker, record):
             )
 
             try:
+                before_build.send(
+                    sender=self.version,
+                    environment=self.build_env,
```

Comment:
We don't have the `environment` var here, I didn't refactor `build_env` (there are a lot of places that depend on the order call), better to refactor it with https://github.com/readthedocs/readthedocs.org/issues/6452 instead of passing it around each method.